src/java.base/share/classes/java/lang/doc-files/ValueBased.html

```diff
@@ -35,32 +35,37 @@ <h1 id="ValueBased">{@index "Value-based Classes"}</h1>
 <code>java.time.LocalDateTime</code>, are <em>value-based</em>.  Instances of a
 value-based class:
 <ul>
-    <li>are final and immutable (though may contain references to mutable
+    <li>are final</li>
+    <li>have only final instance fields (though these may contain references to mutable
         objects);</li>
+    <li>extend either <code>Object</code> or a hierarchy of abstract classes that
+        declare no instance fields and have no instance initialization logic</li>
```

Comment:
This is new, and is relevant only to suitability for inline class migration. But it's true for all classes that claim to be value-based, so seems like a reasonable thing to state, and a good way to encourage people who care about this concept to structure their API in a migration-friendly way.